manifests/params.pp

```diff
@@ -179,13 +192,16 @@
 
   # Passenger config
   $server_app_root = "${dir}/rack"
-  $server_ssl_dir  = "${vardir}/ssl"
+  $server_ssl_dir  = $ssldir
 
   $server_package     = undef
   $server_version     = undef
-  $client_package     = $::osfamily ? {
-    'Debian' => ['puppet-common','puppet'],
-    default  => ['puppet'],
+  $client_package     = versioncmp($::puppetversion, '4.0') ? {
+    '-1'     => $::osfamily ? {
+      'debian' => ['puppet-common','puppet'],
+      default   => ['puppet'],
+    },
+    default => 'puppet-agent',
```

Comment:
I wonder if we should adapt the ```puppet::global``` pattern where you can override the AIO/PE style and then the OS checks come in. Otherwise you need to set parameters in init.pp if a user overrides them. Or would you just have a variable in params.pp for now?